core/src/test/scala/unit/kafka/utils/TestUtils.scala

```diff
@@ -885,6 +891,24 @@ object TestUtils extends Logging {
     ), "Timed out waiting for broker metadata to propagate to all servers", timeout)
   }
 
+  /**
+   * Wait until the expected number of partitions is in the metadata cache in each broker.
+   *
+   * @param servers The list of servers that the metadata should reach to
+   * @param topic The topic name
+   * @param expectedNumPartitions The expected number of partitions
+   */
+  def waitUntilMetadataIsPropagatedWithExpectedSize(servers: Seq[KafkaServer], topic: String, expectedNumPartitions: Int): Unit = {
```

Comment:
What do you think about more concise names?
`waitUntilMetadataIsPropagatedWithExpectedSize` -> `waitForAllPartitionMetadata`
`waitUntilMetadataIsPropagated` -> `waitForPartitionMetadata`

I wonder if this would be more useful if we return the partition metadata: `Map[TopicPartition, UpdateMetadataPartitionState`. Then we could probably skip the calls to `waitUntilMetadataIsPropagated` and `waitUntilLeaderIsElectedOrChanged` above.

Reply:
Good suggestion. Updated.